flink-ml-python/pyflink/ml/recommendation/swing.py

```diff
@@ -0,0 +1,209 @@
+        ParamValidators.gt_eq(0))
+
+    BETA: Param[float] = FloatParam(
+        "beta",
+        "This parameter is used to calculate weight of each user.",
+        0.3,
+        ParamValidators.gt_eq(0))
+
+    def __init__(self, java_params):
+        super(_SwingParams, self).__init__(java_params)
+
+    def set_user_col(self, value: str):
+        return typing.cast(_SwingParams, self.set(self.USER_COL, value))
+
+    def get_user_col(self) -> str:
+        return self.get(self.USER_COL)
+
+    def set_item_col(self, value: str):
+        return typing.cast(_SwingParams, self.set(self.ITEM_COL, value))
+
+    def get_item_col(self) -> str:
+        return self.get(self.ITEM_COL)
+
+    def set_k(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.K, value))
+
+    def get_k(self) -> int:
+        return self.get(self.K)
+
+    def set_max_user_num_per_item(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.MAX_USER_NUM_PER_ITEM, value))
+
+    def get_max_user_num_per_item(self) -> int:
+        return self.get(self.MAX_USER_NUM_PER_ITEM)
+
+    def set_min_user_behavior(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.MIN_USER_BEHAVIOR, value))
+
+    def get_min_user_behavior(self) -> int:
+        return self.get(self.MIN_USER_BEHAVIOR)
+
+    def set_max_user_behavior(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.MAX_USER_BEHAVIOR, value))
+
+    def get_max_user_behavior(self) -> int:
+        return self.get(self.MAX_USER_BEHAVIOR)
+
+    def set_alpha1(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.ALPHA1, value))
+
+    def get_alpha1(self) -> int:
+        return self.get(self.ALPHA1)
+
+    def set_alpha2(self, value: int):
+        return typing.cast(_SwingParams, self.set(self.ALPHA2, value))
+
+    def get_alpha2(self) -> int:
+        return self.get(self.ALPHA2)
+
+    def set_beta(self, value: float):
+        return typing.cast(_SwingParams, self.set(self.BETA, value))
+
+    def get_beta(self) -> float:
+        return self.get(self.BETA)
+
+    @property
+    def user_col(self) -> str:
+        return self.get_user_col()
+
+    @property
+    def item_col(self) -> str:
+        return self.get_item_col()
+
+    @property
+    def k(self) -> int:
+        return self.get_k()
+
+    @property
+    def max_user_num_per_item(self) -> int:
+        return self.get_max_user_num_per_item()
+
+    @property
+    def min_user_behavior(self) -> int:
+        return self.get_min_user_behavior()
+
+    @property
+    def max_user_behavior(self) -> int:
+        return self.get_max_user_behavior()
+
+    @property
+    def alpha1(self) -> int:
+        return self.get_alpha1()
+
+    @property
+    def alpha2(self) -> float:
+        return self.get_alpha2()
+
+    @property
+    def beta(self) -> float:
+        return self.get_beta()
+
+
+class Swing(JavaRecommendationTransformer, _SwingParams):
```

Comment:
In Java code, `Swing` is an implementation of `AlgoOperator`. So here `Swing ` should extends corresponding Python class.